torch/csrc/jit/backends/backend_detail.cpp

```diff
@@ -48,6 +52,221 @@ c10::FunctionSchema getExecuteSchema() {
       /*arguments=*/{self, handle, input},
       /*returns=*/{output});
 }
+
+Module codegen_backend_module(const std::string& backend_name,
+                              const Module& orig_module,
+                              const c10::Dict<IValue, IValue>& method_compile_spec,
+                              c10::DictTypePtr any_dict_ty) {
+  const c10::QualifiedName qual_backend_name(
+      {"__torch__",
+       "torch",
+       "classes",
+       detail::kBackendsNamespace,
+       backend_name});
+  // TODO: Validate method_compile_spec.
+
+  // Clone orig_module to make sure backend transformation is
+  // functional.
+  auto cloned_module = orig_module.clone();
+
+  // Generate LoweredModule.
+  Module loweredModule(
+      "torch.jit." + backend_name + "LoweredModule",
+      get_python_cu(),
```

Comment:
We need to get rid of this. I think this is the only remaining dependency on Python. Unfortunately, this returns a pointer to an object that is defined in Python. For the purposes of the experiment, I think we can add another parameter to the function of type `std::shared_ptr<CompilationUnit>`. In Python, the caller should pass `get_python_cu()`. In C++, the caller will have to use a compilation unit created by loading a module (see source code for `torch.jit.save`) or make their own.